### ISSUE TITLE
Hw2 issues

### ISSUE THREAD

**Issue by LeonidKm** [kommbatsoon/leverx_Ungur_sapui5_homework02#1]
Nice implementation!
1)    printOrderInfo('001') Usually hard coded values are not good and if Orders array will differ you'll have a problem here. Consider getting first order from array and using it's id.
2) printOrders method 
you're using inline handler `onclick="printOrderInfoNext('+order.id+')">'+` that's  kind of bad practice. Please use js attach methods so it'll be easier to find it and debug later. In this method I would recommend getting order Id from event.target ( similar to your code here 
`    var searchValue = event.target.value.toLowerCase();
` )

3) You are using printOrderInfoNext() method handling click on the left part order item.  I would recommend using onOrderItemPress method, that will:
- get id from event target object
- find this object from array of objects
- set this object as variable like selectedOrder so it could be used later for other methods
- do logic for the right part creation(already in place)
Also this method is very big right now(almost 100 lines) so it could be separated to smaller methods
 
4)searchProducts method - you can get current order id from selectedOrder  variable from previous point. Or you can set it anywhere else but anyway consider removing it from inline methods and removing inline methods also. In that case there'll be no need in creation of the table header line each time. 
 
![image](https://user-images.githubusercontent.com/22882321/36148507-6449cb8c-10cd-11e8-91c6-952474875ff1.png)
Repeating code. In case of need to add one more column you'll have to change it in 4 places. Please consider moving this part to function.
5) tabletHead - used in many functions 
![image](https://user-images.githubusercontent.com/22882321/36148690-f4095e68-10cd-11e8-8950-ce72cc200242.png)
I guess it could be removed completely as it's not changing. But if it's needed for you - you can create method createTableHeader that will return it  to you.

6) 
![image](https://user-images.githubusercontent.com/22882321/36148805-5623eb90-10ce-11e8-9788-bb4fb478e6c6.png)
you've already defined 
`    searchProduct = document.getElementById('searchProduct'),
`
at the beginning. So you can use it.

7)`//Print one order
function printOrders(order) {`  confusing naming for the method cause it prints one order, but sounds like it takes and orders array and prints them all.

However everything works, good job!

**Comment by kommbatsoon** [comment]
Thanks for your comments, all the shortcomings will be modified. 
But did you see my recursive method? 
If not, please open console and check it.

**Comment by LeonidKm** [comment]
Yep, saw it, looks good. However you could move object[key] to variable like currentProperty. But I've mentioned similar points in the main code reveiw.

**Comment by kommbatsoon** [comment]
Understand, thanks.

**Comment by kommbatsoon** [comment]
All the shortcomings have been fixed. Are there some other comments to make my code better?

**Comment by LeonidKm** [comment]
In general looks good, let's keep it as is and focus on next versions.
